src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import {Chip} from "../src/Chip.sol";
+
+contract EVMBlackjack {
+    /*//////////////////////////////////////////////////////////////
+    //  Events
+    //////////////////////////////////////////////////////////////*/
+
+    event BetPlaced(address indexed player, uint16 betSize);
+    event PlayerCardDealt(address indexed player, uint8 card, uint8 handIndex);
+    event DealerCardDealt(address indexed player, uint8 card);
+    event InsuranceTaken(address indexed player, bool take);
+    event PlayerActionTaken(address indexed player, Action action);
+    event PlayerBust(address indexed player);
+    event DealerBust(address indexed player);
+    event PayoutsHandled(address indexed player, uint16 playerPayout, uint16 dealerPayout);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Errors
+    //////////////////////////////////////////////////////////////*/
+
+    error InvalidStateTransition();
+    error InvalidBetSize(uint16 betSize);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Data Structures
+    //////////////////////////////////////////////////////////////*/
+
+    enum State {
+        READY_FOR_BET,
+        WAITING_FOR_RANDOMNESS,
+        READY_FOR_INSURANCE,
+        READY_FOR_PLAYER_ACTION,
+        DEALER_ACTION,
+        PAYOUTS
+    }
+
+    enum Action {
+        NO_ACTION,
+        SPLIT_ACES,
+        SPLIT,
+        DOUBLE_DOWN,
+        HIT,
+        STAND
+    }
+
+    struct Game {
+        State state;
+        uint16 shoeCount;
+        uint8[] dealerCards;
+        uint16 insurance;
```

Comment:
assuming that you've pulled this thread already, given this field's placement, but is insurance taken per-hand or per-game? why uint16 and not a boolean?

Reply:
Per round of play, and due to betSize on initial hand being able to be changed, I wanted to set insurance to an actual amount (50% of initial betSize) vs a Boolean, bc later the initial betSize may have changed and an incorrect calculation happen.

Reply:
nice; good call